back/app/controllers/omniauth_callback_controller.rb

```diff
@@ -183,21 +183,17 @@ def set_auth_cookie(provider: nil)
   # @param [OmniauthMethods::Base] authver_method
   # @param [User] user
   def update_user!(auth, user, authver_method)
-    return if authver_method.updateable_user_attrs.empty?
-
     attrs = authver_method.updateable_user_attrs
-    update_hash = authver_method.profile_to_user_attrs(auth).slice(*attrs).compact
+    # If password_login is disabled, users cannot update their emails on UI,
+    # but we still want to keep their emails up to date.
+    attrs << :email if !AppConfiguration.instance.feature_activated?('password_login') && attrs.exclude?(:email)
+    return if attrs.empty?
+
+    update_hash = authver_method.profile_to_user_attrs_for_update(auth).slice(*attrs).compact
     update_hash.delete(:remote_avatar_url) if user.avatar.present? # don't overwrite avatar if already present
     user.confirm! if authver_method.email_confirmed?(auth) # confirm user email if not already confirmed
 
-    if authver_method.overwrite_user_attrs?
```

Comment:
We added `overwrite_user_attrs` to handle empty name in Vienna SSO response and so to avoid overwriting manually set values https://github.com/CitizenLabDotCo/citizenlab/pull/3024

Now I do it with `profile_to_user_attrs_for_update`.